File: python/apcs/1/h026.py
f=int(input())
input()
l=[-1]+[int(x) for x in input().split()]
h=[]
k=0
flag=False
for i in l[1:]:
    k+=1
    h.append(f)
    if f==0:
        if i==2:
            print(*h[:k+1],': Won at round',k)
            flag=True
            break
        elif i==5:
            print(*h[:k+1],': Lost at round',k)
            flag=True
            break
    elif f==2:
        if i==5:
            print(*h[:k+1],': Won at round',k)
            flag=True
            break
        elif i==0:
            print(*h[:k+1],': Lost at round',k)
            flag=True
            break
    else:
        if i==0:
            print(*h[:k+1],': Won at round',k)
            flag=True
            break
        elif i==2:
            print(*h[:k+1],': Lost at round',k)
            flag=True
            break
    if l[k]==l[k-1]:
        if i==0:
            f=5
        elif i==2:
            f=0
        else:
            f=2
if flag==False:
    print(*h,': Drew at round',len(h))

File: cpp/apcs/1/h026/h026.cpp
#include<iostream>
using namespace std;
int main(){
    int f,n,c=0;
    string flag="Drew";
    cin>>f;
    cin>>n;
    int l[10], fl[10];
    for(int i=0;i<n;i++){
        cin>>l[i];
    }
    for(int i=0;i<n;i++){
        c++;
        fl[i]=f;
        if(l[i]==0){
            if(f==2){
                flag="Lost";
                break;
            }
            else if(f==5){
                flag="Won";
                break;
            }
        }
        else if(l[i]==2){
            if(f==0){
                flag="Won";
                break;
            }
            else if(f==5){
                flag="Lost";
                break;
            }
        }
        else if(l[i]==5){
            if(f==0){
                flag="Lost";
                break;
            }
            else if(f==2){
                flag="Won";
                break;
            }
        }
        if(i>=1){
            if(l[i]==l[i-1]){
                if (l[i]==0){
                    f=5;
                }
                else if(l[i]==2){
                    f=0;
                }
                else{
                    f=2;
                }
            }
            else{
                f=l[i];
            }
        }
        else{
            f=l[i];
        }
    }
    for(int i=0;i<c;i++){
        cout<<fl[i]<<" ";
    }
    cout<<": "<<flag<<" at round "<<c;
    return 0;

}